src/main/java/ann/storage/Storage.java

```diff
@@ -0,0 +1,156 @@
+                throw new StorageOperationException("Error opening folder: " + folderName);
+            }
+        }
+        this.folderName = folderName;
+    }
+
+    /**
+     * Create a new file with the given filename in the given folder.
+     *
+     * @param folderName a String which is the name of the folder.
+     * @param fileName a String which is the name of the file.
+     * @throws StorageOperationException if the file cannot be created.
+     */
+    private void openFile(String folderName, String fileName) throws StorageOperationException {
+        Path filePath = Paths.get(folderName + fileName);
+        if(!Files.exists(filePath)) {
+            file = new File(folderName, fileName);
+            try {
+                file.createNewFile();
+            } catch (IOException e) {
+                throw new StorageOperationException("Error opening file: " + fileName);
+            }
+        }
+        this.fileName = fileName;
+    }
+
+    /**
+     * Returns an ArrayList of Strings which represent the Tasks, read from the storage file.
+     *
+     * @return an ArrayList of Strings which represent the Tasks.
+     * @throws AssertionError
+     */
+    public ArrayList<String> load() throws AssertionError{
+        ArrayList<String> tasks = new ArrayList<>();
+        try {
+            Scanner sc = new Scanner(file);
+            while (sc.hasNext()) {
+                tasks.add(sc.nextLine());
+            }
+            return tasks;
+        } catch (FileNotFoundException e) {
+            throw new AssertionError("A non-existent file scenario is already handled earlier");
+        }
+    }
+
+    /**
+     * Writes to the storage file the description of all the tasks in the current TaskList.
+     *
+     * @param taskList the current TaskList of the program.
+     * @throws StorageOperationException if unable to write to file.
+     */
+    public void save(TaskList taskList) throws StorageOperationException {
+        try {
+            String content = taskList.getFileString();
+            FileWriter fw = new FileWriter(file);
+            fw.write(content);
+            fw.close();
+        } catch (IOException e) {
+            throw new StorageOperationException("Error writing to file: " + this.folderName + "/" + this.fileName);
+        }
+    }
+
+    /**
+     * Returns a boolean which indicates whether the given filename is valid.
+     *
+     * @param fileName a String which is the filename.
+     * @return true if the given filename ends with '.txt' and false otherwise.
+     */
+    private boolean isValidFileName(String fileName) {
+        return fileName.endsWith(".txt");
+    }
+
+    /**
+     * Represents an Exception thrown when a given filename is not valid.
+     */
+    public static class InvalidStorageFilePathException extends IllegalArgumentException {
+        public InvalidStorageFilePathException(String message) {
+            super(message);
+        }
+    }
```

Comment:
I like that this particular exception extends from `IllegalArgumentException`! Since it also represents some form of invalid argument, and this exception also allows clearer error handling with regards to the Storage class for your AnnBot!